### ISSUE TITLE
Restore resnet-101 model pretrained on imagenet

### ISSUE THREAD

**Issue by j50888** [DrSleep/tensorflow-deeplab-resnet#63]
Hi,
As title,I want to initial the model(only the same architecture layer) with resnet-101 pretrained on imagenet
I tried to set the differnet layer to NOT_RESTORE_VARIABLES
But it got errors like below:
`NotFoundError (see above for traceback): Tensor name "bn4b16_branch2c/moving_mean" not found in checkpoint files ./resnet_v1_101.ckpt
         [[Node: save_1/RestoreV2_190 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](_recv_save_1/Const_0, save_1/RestoreV2_190/tensor_names, save_1/RestoreV2_190/shape_and_slices)]]
         [[Node: save_1/RestoreV2_233/_761 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/gpu:0", send_device="/job:localhost/replica:0/task:0/cpu:0", send_device_incarnation=1, tensor_name="edge_3148_save_1/RestoreV2_233", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:0"]()]]
`
That was due to the difference of layer names or something else?
BTW,is there some faster approach to restore this kind of models?

**Comment by DrSleep** [comment]
First, it would be helpful to know where is the file `./resnet_v1_101.ckpt` coming from. The error might be due to a different name of the `moving_mean' tensor, or just simple because `moving_mean` is not present in your ckpt file.

**Comment by j50888** [comment]
I downlaod from this link
[https://github.com/tensorflow/models/tree/master/slim](url)
How could I to check the name of that variable in the ckpt file?
Or I should trace the source code which generate it?

**Comment by DrSleep** [comment]
For example, you can use something like this (adapted from [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/tools/inspect_checkpoint.py#L32)):
```python
from tensorflow.python import pywrap_tensorflow
reader = pywrap_tensorflow.NewCheckpointReader('./resnet_v1_101.ckpt')
var_to_shape_map = reader.get_variable_to_shape_map()
for key in var_to_shape_map:
    print(key)
```

But as written in the Pre-trained models section,

> Note that the VGG and ResNet parameters have been converted from their original caffe formats (here and here)

you might as well just convert the corresponding caffemodel file similar to the instructions in README of this repository: https://github.com/DrSleep/tensorflow-deeplab-resnet#caffe-to-tensorflow-conversion

**Comment by j50888** [comment]
Hi,
I tried to convert the caffe pretrained-model to tensorflow checkpoint model.
But the convert.py have problem when converting ResNet model [here](https://github.com/ethereon/caffe-tensorflow/issues/104)
So I used the [tensorpack](https://github.com/ppwwyyxx/tensorpack/tree/master/examples/ResNet) to convert the caffe model to npy file.
Then I used the npy2ckpt.py.
But I encountered this error : 
`WARNING:tensorflow:From npy2ckpt.py:58 in main.: initialize_all_variables (from tensorflow.python.ops.variables) is deprecated and will be removed after 2017-03-02.
Instructions for updating:
Use `tf.global_variables_initializer` instead.
Traceback (most recent call last):
  File "npy2ckpt.py", line 69, in <module>
    main()
  File "npy2ckpt.py", line 62, in main
    net.load(args.npy_path, sess)
  File "/tmp3/haowei/tensorflow-deeplab-resnet/kaffe/tensorflow/network.py", line 63, in load
    for param_name, data in data_dict[op_name].iteritems():
AttributeError: 'numpy.ndarray' object has no attribute 'iteritems'
`
Is the problem occur because of the npy file?
